packages/gatsby-source-contentful/src/report.js

```diff
@@ -0,0 +1,31 @@
+export const CODES = {
+  /* Fetch errors */
+
+  LocalesMissing: `111001`,
+  SelfSignedCertificate: `111002`,
+  SyncError: `111003`,
+  FetchContentTypes: `111004`,
+}
+
+export const ERROR_MAP = {
+  [CODES.LocalesMissing]: {
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `USER`,
+  },
+  [CODES.SelfSignedCertificate]: {
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `USER`,
+  },
+  [CODES.SyncError]: {
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `THIRD_PARTY`,
+  },
+  [CODES.FetchContentTypes]: {
+    text: context => context.sourceMessage,
+    level: `ERROR`,
+    category: `THIRD_PARTY`,
+  },
+}
```

Comment:
This is where I most need feedback from the plugin authors. By using these codes with this error map, we want to keep the following things in mind:

1. Codes should be unique, meaningful, maintainable, and searchable.
1. The map should indicate the correct error category & level.

I made my best guess at assigning meaningful code names and categories and invite other contributors to offer corrections.

Reply:
I think this looks correct. I'm going to sign off on it but if @axe312ger has further thoughts we can fix forward.